src/plugin/google/places/attrcard.js

```diff
@@ -1,48 +1,61 @@
-goog.provide('plugin.google.places.AttrCardCtrl');
-goog.provide('plugin.google.places.attrCardDirective');
+goog.module('plugin.google.places.AttrCardUI');
 
-goog.require('os');
-goog.require('os.ui.Module');
+const {ROOT} = goog.require('os');
+const Module = goog.require('os.ui.Module');
 
 
 /**
  * The geonames result card directive for displaying search results.
  *
  * @return {angular.Directive}
  */
-plugin.google.places.attrCardDirective = function() {
-  return {
-    replace: true,
-    restrict: 'E',
-    templateUrl: os.ROOT + 'views/plugin/google/places/attrcard.html',
-    controller: plugin.google.places.AttrCardCtrl,
-    controllerAs: 'attrCtrl'
-  };
-};
+const directive = () => ({
+  replace: true,
+  restrict: 'E',
+  templateUrl: ROOT + 'views/plugin/google/places/attrcard.html',
+  controller: Controller,
+  controllerAs: 'attrCtrl'
+});
+
+/**
+ * The element tag for the directive.
+ * @type {string}
+ */
+const directiveTag = 'googleplacesattrcard';
 
 
 /**
  * Register the directive.
  */
-os.ui.Module.directive('googleplacesattrcard', [plugin.google.places.attrCardDirective]);
+Module.directive('googleplacesattrcard', [directive]);
 
 
 
 /**
- * @param {!angular.Scope} $scope
- * @param {!angular.JQLite} $element
- * @constructor
- * @ngInject
+ * @unrestricted
  */
-plugin.google.places.AttrCardCtrl = function($scope, $element) {
-  // just plop the result HTML in
-  var attrs = /** @type {plugin.google.places.AttrResult} */ ($scope['result']).getResult();
-
-  var html = '';
-  for (var i = 0, n = attrs.length; i < n; i++) {
-    html += '<div>' + attrs[i] + '</div>';
+class Controller {
+  /**
+   * Constructor.
+   * @param {!angular.Scope} $scope
+   * @param {!angular.JQLite} $element
+   * @ngInject
+   */
+  constructor($scope, $element) {
+    // just plop the result HTML in
+    var attrs = /** @type {plugin.google.places.AttrResult} */ ($scope['result']).getResult();
```

Comment:
Not sure if we added scripts for global replacement of types, but we can come back on these later.

Reply:
Type-only references are tricky to detect because comments are simply text in AST. You can sort of do a regex detect, but that gets ugly fast with all the variations you might see in `@type` or `@param` annotations with `|`, `?`, etc. I think there is a library to parse out JSDoc comments in a way that would make this easier, but I haven't really looked into it. Currently types in comments are only replaced if the type is actually referenced in code where we can detect it with AST.